src/util/locking/scripts/RedisLuaScripts.ts

```diff
@@ -0,0 +1,151 @@
+    if ((redis.call("exists", lockKey) == 1) or (count ~= nil and count > 0)) then
+      return 0
+    end
+    
+    -- Set lock and respond with 'OK' if succeeded (otherwise null)
+    return redis.call("set", lockKey, "${LOCKED}");
+    `,
+  releaseReadLock: `
+      -- Return 1 after decreasing the counter, if counter is < 0 now: return '-ERR'
+      local countKey = KEYS[1].."${SUFFIX_COUNT}"
+      local result = redis.call("decr", countKey)
+      if result >= 0 then
+        return 1
+      else 
+        return redis.error_reply("Error trying to release readlock when read count was 0.")
+      end
+    `,
+  releaseWriteLock: `
+      -- Release the lock and reply with 1 if succeeded (otherwise return '-ERR')
+      local lockKey = KEYS[1].."${SUFFIX_WLOCK}"
+      local result = redis.call("del", lockKey)
+      if (result > 0) then
+        return 1
+      else
+        return redis.error_reply("Error trying to release writelock that did not exist.")
+      end
+    `,
+  acquireLock: `
+      -- Return 0 if lock entry already exists, or 'OK' if it succeeds in setting the lock entry.
+      local key = KEYS[1].."${SUFFIX_LOCK}"
+      if redis.call("exists", key) == 1 then
+        return 0
+      end
+      
+      -- Return 'OK' if succeeded setting entry
+      return redis.call("set", key, "${LOCKED}");
+      `,
+  releaseLock: `
+      -- Release the lock and reply with 1 if succeeded (otherwise return '-ERR')
+      local key = KEYS[1].."${SUFFIX_LOCK}"
+      local result = redis.call("del", key)
+      if result > 0 then
+        return 1
+      else
+        return redis.error_reply("Error trying to release lock that did not exist.")
+      end
+    `,
+};
+
+export type RedisAnswer = number | null | 'OK' | string;
+
+/**
+ * Convert a RESP2 response to a boolean.
+ * @param result - The Promise-wrapped result of a RESP2 Redis function.
+ * @returns * `1`, `'OK'`: return `true`
+ *          * `0`: returns `false`
+ *          * `-ERR`: throw error
+ * @throws On `-ERR*` `null` or any other value
+ */
+export async function fromResp2ToBool(result: Promise<RedisAnswer>): Promise<boolean> {
+  const res = await result;
+  switch (res) {
+    case 1:
+    case 'OK':
+      return true;
+    case 0:
+      return false;
+    case null:
+      throw new Error('Redis operation error detected (value was null).');
+    default:
+      if (res.toString().startsWith('-ERR')) {
+        throw new InternalServerError(`Redis error: ${res.toString().slice(5)}`);
+      } else {
+        throw new Error(`Unexpected Redis answer received! (${res})`);
+      }
+  }
+}
+
+export interface RedisReadWriteLock extends RedisInterface {
+  /**
+     * Try to acquire a readLock on `resourceIdentifierPath`.
+     * Will succeed if there are no write locks.
+     *
+     * @returns 1 if succeeded. 0 if not possible.
+     */
+  acquireReadLock: (resourceIdentifierPath: string, callback?: Callback<string>) => any;
```

Comment:
```suggestion
  acquireReadLock: (resourceIdentifierPath: string, callback?: Callback<string>) => RedisAnswer;
```
Is this correct? Or even more specific if that is too broad. Same for all the ones below.